docs/bots-guide.md

```diff
@@ -0,0 +1,230 @@
+
+ * In the *Your Bots* panel, click on the green icon to download its configuration file *.zuliprc*
+ (the structure of this file is explained [here](#configuration-file)
+
+ * Copy the file to a destination of your choice on your Zulip server, e.g. to `~/.zuliprc` or `~/zuliprc-test`.
+
+5. Subscribe the bot to any streams that the bot needs to read messages from or write messages to.
+ * To subscribe your bot to streams, navigate to *Manage Streams*. Select a stream and
+ add your bot by its email address (the address you assigned in step 3).
+
+ * Now, the bot will do its job on the streams you subscribed it to.
+
+6. Run the bot.
+ * On your Zulip server (and outside the Vagrant environment), navigate to `~/zulip/contrib_bots/`
+
+ * Run `python run.py ~/zulip/contrib_bots/<my-bot>.py --config-file ~/.zuliprc`. The `~/` before `.zuliprc`
+ should point to the directory containing the file (in this case, it is the home directory).
+
+ * Check the output of the command. It should start with the text the `usage` function returns,
+ followed by logging output similar to this:
+```
+INFO:root:starting message handling...
+INFO:requests.packages.urllib3.connectionpool:Starting new HTTP connection (1): localhost
+```
+* Congrats! Now, your bot should be ready to test on the streams you've subscribed it to.
+
+### Test the `followup.py` bot
+1. Do the previous steps for the `followup.py` bot.
+2. Create the *followup* stream.
+3. Subscribe the bot to the newly created *followup* stream and a stream you want to use it from, e.g. *social*.
+4. Send a message to the stream you've subscribed the bot to (other than *followup*). If everything works, a copy of the message should now pop up in the *followup* stream.
+
+## How to develop a bot
+
+The tutorial below explains the structure of a bot `<my-bot>.py`. You can use this as boilerplate code for developing your own bot.
+
+Every bot is built upon this structure:
+```
+class FollowupHandler(object):
+    '''
+    A docstring documenting this bot.
+    '''
+
+    def usage(self):
+        return '''Your description of the bot'''
+
+    def triage_message(self, message):
+        #add your code here
+
+    def handle_message(self, message, client, state_handler):
+        # add your code here
+
+handler_class = FollowupHandler
+```
+* The class name (in this case *FollowupHandler*) can be defined by you and should match the name of your bot. To register your bot's class, adjust the last line `handler_class = FollowupHandler` to match your class name.
+
+* Every bot needs to implement the functions
+ * `usage(self)`
+ * `triage_message(self, message)`
+ * `handle_message(self, message, client)`
+
+* These functions are documented in the [next section](#bot-API).
+
+## Bot API
+This section documents the functions every bot needs to implement and the structure of the bot's config file.
+
+### usage
+*usage(self)*
+Is called to retrieve information about the bot.
```

Comment:
If you want "Is called to retrieve information about the bot." to start on a new line, you need to insert another line break. Same goes for triage_message, handle_message, etc.